tools/topology/topology2/platform/intel/speaker-echo-ref.conf

```diff
@@ -33,41 +13,24 @@ Object.Pipeline {
 				copier_type	"SSP"
 				stream_name	$SPEAKER_CODEC_NAME
 				node_type	$I2S_LINK_INPUT_CLASS
-				Object.Base.audio_format.1 {
-					in_bit_depth		32
-					in_valid_bit_depth	32
-					out_bit_depth		32
-					out_valid_bit_depth	32
-				}
+				num_input_audio_formats 1
+				Object.Base.input_audio_format [
+					{
+						in_bit_depth		32
+						in_valid_bit_depth	32
+					}
+				]
+				num_output_audio_formats 1
+				Object.Base.output_audio_format [
+					{
+						out_bit_depth		32
+						out_valid_bit_depth	32
+					}
+				]
 			}
 			Object.Widget.pipeline."1" {
 				core	$ECHO_REF_CORE_ID
 			}
 		}
 	]
 }
-
-Object.PCM.pcm [
-	{
-		name	"EchoRef"
-		id	$ECHO_REF_PCM_ID
-		direction	"capture"
-		Object.Base.fe_dai.1 {}
-
-		Object.PCM.pcm_caps.1 {
-			name	$ECHO_REF_STREAM_NAME
-			formats	'S16_LE,S24_LE,S32_LE'
-		}
-	}
-]
-
-Object.Base.route [
-	{
-		source	"dai-copier.SSP.$SPEAKER_CODEC_NAME.capture"
-		sink	$ECHO_REF_COPIER_MODULE
-	}
-	{
-		source	$ECHO_REF_COPIER_MODULE
-		sink	"host-copier.$ECHO_REF_PCM_ID.capture"
-	}
-]
```

Comment:
@ranj063  Not necessarily for this PR, but should we still have templates for both cases, echo ref to an AEC running on the DSP (like this case), and echo ref to user-space. I mean a given product will use only one option, not both, but in the past we've enabled both.

Reply:
@kv2019i for aec in userspace, we would need the host copier and the PCM but we can't tie it to this topology. So I'd say we should leave it until.theres a need for it.